database-commons/src/main/java/co/cask/db/batch/action/DBRun.java

```diff
@@ -41,28 +42,24 @@ public DBRun(QueryConfig config, Class<? extends Driver> driverClass) {
    * to use and which connection string to use come from the plugin configuration.
    */
   public void run() throws SQLException, InstantiationException, IllegalAccessException {
-    DBManager dbManager = new DBManager(config);
-
+    DriverCleanup driverCleanup = null;
     try {
-      dbManager.ensureJDBCDriverIsAvailable(driverClass);
+      driverCleanup = DBUtils.ensureJDBCDriverIsAvailable(driverClass, config.getConnectionString(),
+                                                          config.jdbcDriverName);
 
       try (Connection connection = getConnection()) {
-        if (!config.enableAutoCommit) {
-          connection.setAutoCommit(false);
-        }
         try (Statement statement = connection.createStatement()) {
           statement.execute(config.query);
-          if (!config.enableAutoCommit) {
-            connection.commit();
-          }
         }
       }
     } finally {
-      dbManager.destroy();
+      if (driverCleanup != null) {
+        driverCleanup.destroy();
+      }
     }
   }
 
   private Connection getConnection() throws SQLException {
```

Comment:
Not from your code, but let's remove this one line method and just call DriverManager.getConnection() directly.

Reply:
done